src/App.tsx

```diff
@@ -1,223 +1,190 @@
+/* eslint-disable no-console */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
-import { AuthContext } from './components/Auth/AuthContext';
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  TodoData,
+} from './api/todos';
+import { AuthContext, AuthProvider } from './components/Auth/AuthContext';
+// eslint-disable-next-line max-len
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotiication';
+import { NewTodoForm } from './components/NewTodoForm/NewTodoForm';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { TodoList } from './components/TodoList/TodoList';
+import { FilterOptions } from './types/FilterOptions';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState('');
+  const [isErrorHidden, setIsErrorHidden] = useState(true);
+  const [selectedOption, setSelectedOption] = useState(FilterOptions.ALL);
+  const [isAdding, setIsAdding] = useState(false);
+  const [deletedTodoIds, setDeletedTodoIds] = useState<number[]>([]);
+
+  const getUserTodos = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch (err) {
+        setError('get');
+        setIsErrorHidden(false);
+      }
+    }
+  };
+
+  const addNewTodo = async (todo: TodoData) => {
+    try {
+      setIsErrorHidden(true);
+      setIsAdding(true);
+      const tempTodo = { ...todo, id: 0 };
+
+      setTodos(prevTodos => [...prevTodos, tempTodo]);
+
+      await createTodo(todo);
+    } catch (err) {
+      setError('add');
+      setIsErrorHidden(false);
+    } finally {
+      setIsAdding(false);
+      getUserTodos();
     }
+  };
+
+  const deleteCurrentTodo = async (todoId: number) => {
+    try {
+      setIsErrorHidden(true);
+      setDeletedTodoIds((currentIDs) => [...currentIDs, todoId]);
+
+      await deleteTodo(todoId);
+
+      getUserTodos();
+
+      return 1;
```

Comment:
I don`t think U need to return smth here

Reply:
I need this returns to use this function in `clearCompletedTodos`)))

Reply:
maybe let's show an error instead of returning something?